plugins/plugins.go

```diff
@@ -12,14 +13,16 @@ type Item struct {
 }
 
 type Plugin struct {
-	ID      string
-	Enabled bool
-	Limit   chan struct{}
+	ID    string
+	Limit chan struct{}
+}
+
+type Channels struct {
+	Items     chan Item
+	Errors    chan error
+	WaitGroup *sync.WaitGroup
 }
 
 type IPlugin interface {
-	DefineCommandLineArgs(cmd *cobra.Command) error
-	Initialize(cmd *cobra.Command) error
-	GetItems(chan Item, chan error, *sync.WaitGroup)
-	IsEnabled() bool
+	DefineCommand(channels Channels) *cobra.Command
 }
```

Comment:
@jossef @joaopedrocsilva @cx-monicac I really want your feedback here. You can see the interface of `IPlugin` made thin with just one function for the initialization.

All the logic for a plugin (the flags validation, the plugin fields, and the `GetItems` are under the plugin's responsibility, and the new interface from the plugin run, and the `main` is the `channels`: Errors channel and `Item`s channel.

Reply:
Make sense to me.

Reply:
I agree, with the sub-commands we don't need all this logic anymore

Reply:
Only comment is to add error in response as this is called by the main "command"